synapse/storage/databases/main/state.py

```diff
@@ -62,11 +62,8 @@ async def get_room_version(self, room_id: str) -> RoomVersion:
                 Typically this happens if support for the room's version has been
                 removed from Synapse.
         """
-        return await self.db_pool.runInteraction(
-            "get_room_version_txn",
-            self.get_room_version_txn,
-            room_id,
-        )
+        room_version_id = self.get_room_version_id(room_id)
```

Comment:
```suggestion
        room_version_id = await self.get_room_version_id(room_id)
```
I think this is the reason the tests are failing